### PR TITLE
Move Docker SSI tests to Gitlab

### PR DESCRIPTION
**Pull request by randomanderson**
## Motivation

These tests need to run in Gitlab in order to access non-released versions of the tracer. Those PRs will come later

## Changes
With this PR, the docker ssi tests are a child Gitlab pipeline that generated on every commit. If the pipeline was triggered from a schedule, the tests run automatically. Otherwise the tests run manually.

* Removes the Github workflows and generation logic
* Updates `docker_ssi_matrix_builder.py` to generate a Gitlab pipeline
* Updates the Docker SSI container specs to accept host and port instead of defaulting to localhost
* Uses a random port (`get_free_port()`) rather than a hardcoded port for the agent. In Gitlab, there is already an agent running
* Uses docker commands to determine the correct IPs for containers in Gitlab
* Updates the Jetty app to not throw exceptions on startup (classpath issues)

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
Awesome work Laplie!


```24-11-19 10.19.37:Installing script cli archive...
2024-11-19 10.19.37:* Downloading...
2024-11-19 10.19.37:[91m
2024-11-19 10.19.37:curl: (22) The requested URL returned er[0m
2024-11-19 10.19.37:[91mror: 503 Service Temporarily Unavailabl[0m
2024-11-19 10.19.37:[91me
2024-11-19 10.19.37:[0m
2024-11-19 10.19.37:'curl --fail --location --progress-bar "${SDKMAN_SERVICE}/broker/download/native/install/${SDKMAN_NATIVE_VERSION}/${SDKMAN_PLATFORM}" > "$sdkman_zip_file"': command failed with exit code 22.```

File: utils/tools.py [hunk 1]
```diff
@@ -189,3 +191,17 @@ def nested_lookup(needle: str, heystack, look_in_keys=False, exact_match=False):
         return False
 
     raise TypeError(f"Can't handle type {type(heystack)}")
+
+
+def get_free_port():
+    last_allowed_port = 65535
+    port = random.randint(1100, 65100)
```
[the remark this answers is not in the document]
Comment (randomanderson) [reply utils/tools.py]: The function existed before my PR and I just moved it to a more accessible location.

I lowered the limit to 32000 as suggested

File: utils/docker_ssi/docker_ssi_matrix_builder.py [hunk 2]
```diff
@@ -1,46 +1,91 @@
...
+            "needs": [],
+            "script": ["echo 'DONE'"],
+        },
+        ".base_ssi_job": {
+            "image": "registry.ddbuild.io/ci/libdatadog-build/system-tests:48436362",
+            "needs": [],
+            "script": [
+                "./build.sh -i runner",
+                "source venv/bin/activate",
+                'timeout 2700s ./run.sh DOCKER_SSI --ssi-weblog "$weblog" --ssi-library "$TEST_LIBRARY" --ssi-base-image "$base_image" --ssi-arch "$arch" --ssi-installable-runtime "$installable_runtime" --ssi-force-build',
```
[the remark this answers is not in the document]
Comment (randomanderson) [reply utils/docker_ssi/docker_ssi_matrix_builder.py]: Done

**Comment by randomanderson** [comment]
@robertomonteromiguel sdkman was indeed flaky. They impose a 10/request per second/per IP rate limit. I added sleeps with random timeouts in `java_install_runtimes.sh`. Launching all 52 jobs at once, no issues.

**Comment by robertomonteromiguel** [comment]
@randomanderson The lint failures

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**